### ISSUE TITLE
"Creating the Enemy" Created a Mess

### ISSUE THREAD

**Issue by TesFalcon** [godotengine/godot-docs#8947]
**Your Godot version:**
4.2.1

**Issue description:**
I'm just getting started in Godot. There seem to be a LOT of assumptions being made in the 1st 2D game tutorial. I am LOST.

I went through the player portion, and it was detailed enough that I felt like I understood what was happening. Even better, all my tests worked with no issues.

At the very top of the mob portion (why is it called a mob not squid or enemy?), it seems like the documentation starts leaving out steps and LOTS of explanations.

New Scene? Ok, done.
The new scene includes all the code written for the player scene. The same code is now in both the player tab and the new mod tab. 
Is this 2 different view for the same code or is this 2 different code sections?

RigidBody2D & Child Nodes? Ok, I did that, but I'm not seeing the button to deselect the children like I saw with the player. Where's that at?

Under CollisionObject2D node in the Inspector, there is NO Collision group.
If I search for "mask", I see a mask item as it shows in the picture under "CanvasItem" > "Visibility" which shows (among other things) Light Mask & Visibility Layer. 
I FOUND IT! 
As I was trying to describe this more clearly to you, I saw that the instructions weren't stepping down the tree. It's talking about stuff interchangably on either side of the screen and calling things by their old names rather than using the new name. All these name are VERY similar to each other! Instead, this is supposed to be under the "MOB" node that it keeps calling "RigidBody2D" even though we already changed the name of it. I thought this was under the CollisionShape2D node since it kept talking about the CollisionObject2D section, but the documentation never define what a section is. UGH!

This whole struggle happened again when setting up the collision area. Rather than giving top-down instructions, Transform > Rotation: Set to 90*. It jumps down to the property that needs changing (Rotation) and sets the heading where I'd find it in a parenthetical. (Does anyone even read parenthetical statements?)

My code is "duplicated", but it all saves as the same player.gd script. 
It says add script to the enemy/mob thing of "extends RigidBody2D", etc. 

Do I need to attach a new enemy script or is this code added to the same player script I worked on previously which is already existing under my mod script area? 
Why is there no, "If you did it all correctly, here's what your complete script file will look like."?

When it says to add x code to the _ready() function is that in addition to or to replace the code that's already there?
Does the "pass" command get pushed down to the bottom of the commands for that function as an "end function" action or does it get deleted? The comment says that pass is only there as a place holder until other code is added so that seems to imply that I need to delete pass.

I got particularly frustrated when it came to the last part which states:

"The last piece is to make the mobs delete themselves when they leave the screen. Connect the screen_exited() signal of the VisibleOnScreenNotifier2D node to the Mob and add this code:
GDScript
func _on_visible_on_screen_notifier_2d_screen_exited():
	queue_free()"

I have tried the only 2 ways I know to accomplish this, but it's not working.
When I go to "VisibleOnScreenNotifier2D" to add the signal, it keeps asking me for a valid method. (???)
If I add the GDScript to the mash of code, I can't connect it to the signal. 

Right now this signal is just a huge flashing red light with klaxons.

I'm stumped, stuck, and frustrated.

**URL to the documentation page (if already existing):**
https://docs.godotengine.org/en/stable/getting_started/first_2d_game/04.creating_the_enemy.html

**Comment by skyace65** [comment]
“As I was trying to describe this more clearly to you, I saw that the instructions weren't stepping down the tree. It's talking about stuff interchangably on either side of the screen and calling things by their old names rather than using the new name. All these name are VERY similar to each other! Instead, this is supposed to be under the "MOB" node that it keeps calling "[RigidBody2D](https://docs.godotengine.org/en/stable/classes/class_rigidbody2d.html#class-rigidbody2d)" even though we already changed the name of it. I thought this was under the CollisionShape2D node since it kept talking about the CollisionObject2D section, but the documentation never define what a section is. UGH!”

The tutorial does not call the “MOB” node “RigidBody2D” after you rename it. It’s talking about the RigidBody2D section of the inspector, and it defines a “section” as part of the inspector, the text is clear about this. Here is the first time RigidBody2D and section are mentioned in the page “RigidBody2D section of the inspector”. The next time “RigidBody2D” is referred to in that part of the page it’s still called a section, not a node. In addition the tutorial explicitly tells you to select the “Mob” node then check its RigidBody2D section in the inspector.

“This whole struggle happened again when setting up the collision area. Rather than giving top-down instructions, Transform > Rotation: Set to 90*. It jumps down to the property that needs changing (Rotation) and sets the heading where I'd find it in a parenthetical. (Does anyone even read parenthetical statements?)”

Personally I don’t see an issue with having the section in parenthesis. Especially since you go to the transform section in the previous page for the player node, but a change for this would be fine.

As for your various code issues the player script is not duplicated. The script editor can show all scripts in the project regardless of if they are currently in the scene you have open or not. This should be clarified in the Step by Step section or somewhere else since it’s not currently mentioned. In addition the phrasing for creating a new mob script is definitely a bit odd and should be re-worded

**Comment by TesFalcon** [comment]
I found this YouTube video to explain the 9,000 things that this manual fell down on. This doc SUX! And, skyace65, as a NEW USER and someone just getting back into programming after a 15 year hiatus, or even if I was my 10 yr old son who wanted to learn, your explanation leaves MUCH to be desired. Just because it's obvious and clear to YOU, does NOT make it obvious or clear to anyone else. Why is there a RigidBody2D section and a RigidBody2D node?

While you say that the script is NOT duplicated, you haven't clarified how I'm supposed to handle the coding of the mob. Do I simply add the "extends RigidBody2D" code into the Player code that's there?

You kinda skipped a dozen questions.

**Comment by BastiaanOlij** [comment]
Hi @TesFalcon,

I'm sorry you're having so much difficulty with the introduction series in the Godot manual. There are indeed assumptions that have to be made to keep the manual on point. Many are explained in earlier chapters of the Getting started section of the manual. Sometimes it is a single core element where the description is skipped or unclear, that makes the rest of the documentation hard to follow. 

Getting the right message across in text often ends up being highly subjective, what one group of users find easy to understand, another group of users have trouble with. Sometimes a visual explanation like in the video you linked, works much better then text ever can. As the old adage goes, a picture tells a thousand words. I'd like to extend that to "A video tells a million words". Though even there, we have users who prefer the text version over the video version.

Godot is an open source gaming engine, Godot doesn't have a hired writing team of experts (not that this is a guarantee of good documentation). 
Most of the work is done by volunteers, including by the author of the video you pointed to. These people have put weeks if not months of their free time into making the documentation and spend even more of their free time maintaining it. Many more volunteers such as @skyace65 help out in sifting through these issues to help users out and see what feedback is usable to help improve the documentation.

We are lucky enough to have such a large group of volunteers helping out, and an even larger group willing to make tutorials on Youtube or other places. We understand that getting stuck understanding Godot can be frustrating, but please respect the effort that has been taken here.

I would commend you to take a bit more time to learn Godot, use [the other resources available to you](https://docs.godotengine.org/en/stable/community/tutorials.html), and form a more complete picture of how Godot works. Then come back here and suggest more concrete changes you'd like to see. Instead of "I don't understand this, you did a terrible job, please change it", it becomes "I didn't understand this, I respect the effort that went into this, but if it is written like this new people might have a less difficult time than I did". You might find the needed change to make the proverbial penny drop, is minor, and it all falls into place.

**Comment by TesFalcon** [comment]
Perhaps an introduction is in order.

I joined the USAF at the age of 19 in 1994 and was sent to Keesler AFB for technical training as a Computer Programmer. Up until this point, I'd never programmed a computer before. 

Growing up, we had an old 8088 PC that had a variant of DOS on it. Win3.1 was installed on it, but it was slow, clunky, and none of the programs on it were Windows-based so the extra overhead was unnecessary. The previous owner had written a BATCH program to act as an interface as well as a thick how-to book called "What Is a Computer?" Since staring at a black screen with C:\> and a blinking cursor was a bit unhelpful, I read that book to understand how to use this new machine. While it talked about files systems and executables, there was not a word on how to do any programming. The Internet didn't exist yet so I had no idea that I could have written programs on that old PC.

In Tech School, they taught ADA, Assembler, and BASIC. The intent was to take kids who had never programmed before (like me) and teach us how to code. There were still legacy systems written in ADA that needed to be maintained. Assembler was used for core computer knowledge and CPU processing understanding. The vast majority of the time was spent using BASIC to teach the core concepts of functions, methods, variables, etc.

After Tech, we all got assigned different bases around the world. The vast majority of us were sent to either Alabama or Nebraska. I got Offutt AFB, NE south of Omaha. The programming shop I was assigned to was 100% VB4 on Windows 3.1 PCs. By the end of 1995, we had converted over to Win95, but still running VB4. 

In 1996, all programmers in all of StratCom were required to take an "object oriented programming" course. JAVA had just been released that the Internet was making waves. All our projects written in VB4 were entirely contained on the local Intranet. MOF, instead of using VB4 for my main project, I was using (Classic) ASP written in VB Script, HTML, JavaScript, connected to a SQLServer DB. While I took the course, it had no bearing on my daily life as a developer.

In 1998, I got out the USAF and had options. I was offered 6-figures to do ADA maintenance full-time. I was offered 6-figures to do Y2K work. I didn't think that either of these would have much of a future past the year 2000 so I opted to continue developing in (Classic) ASP and took a Web Developer Analyst role in Redmond, WA. That lasted all of 1999. I missed my Microsoft interview for the SQL Server 7 team by one question at the very end of the interview. In Dec 99, we went to Denver as a family and through a series of bad luck, we got stranded in Denver and was prevented from returning to Washington on time. Boss was ticked off and fired me. Instead of returning to our house, my FIL and I made it a roadtrip to go back to Redmond on Jan 1, 2000 pack up the entire 3 bedroom house by ourselves and take everything to Denver to start over.

In Denver, I worked as Network Administrator / Web Developer (aka Webmaster) for a small start-up of visual artists for ~6 months. Once my project was done, they had no real work for me to do, and I didn't want to stay in Denver anyhow. So I found a contract developer position with the newly created Verizon company in downtown Tampa, FL. That lasted from July 2000 to March of 2001 when the NASDAQ crashed and all the DotComs turned into DotBombs. At Verizon, I worked in VB6 to created a DLL to more easily port an Access DB to SQL Server. I needed it to move the extensive Access DB built up over many years by my customer which only served one office to the SQL Server DB and the new system to serve all of Verizon nation-wide. Another developer in the office needed the same tech so I made sure my DLL was generic enough to run both his project and mine. 

As soon as the project was over, I was out of a job. That was the last time I was ever paid to program. I tried unsuccessfully over the next several months to get a new position in computers doing anything. Nothing. 9/11 was a non-event for me since the frustration of programming circles around my competition that day but still being told that since he had an active project live on the internet and I didn't, he was getting the job. I was totally bummed.

From 2002 - 2005, I did whatever I could to make money, mostly pizza and landscaping. In 2005, I got my CDL and spent 10 years in the oilfield all over the west. Then in 2015 when the oil industry was down (AGAIN!), I started driving semis full-time first OTR then bouncing between regional and local. I still drive "locally". 

In 2015, my HDD crashed, and I had NO computer other than my smartphone until 2021 when I decided to get back into computers and see how the tech had changed. I bought my current PC for $200. I started learning PHP and re-learning the new HTML and JavaScript. I got hosting on a free server to practice what I was learning. I went through each item in my tutorial step-by-step. For whatever reason, several items in the PHP tutorial were forbidden on the server so I coded them, but they never worked. As I reached the end of the tutorial, I reached out to the server folks who had been following my progress and asked about where I should turn my attention next. Their admin proceeded to hack my entire project and trash everything I had struggled so hard with. His only response was, "If you can't keep a hacker like me out then you need to toss your computer into the trash and stop trying." Mind you, this guy was in his mid-20s and lived and breathed computers since he was 5 when he FIRST got placed on house-arrest for hacking.

Fast forward to 2023. I've been enjoying a particular game on my phone for several years. I see where they've made certain design choices that have screwed them over and have made the game less capable than it could have been. I decided to make my own version of the game. In Jan 2024, I did some research on game engines that I could use and all the reviews pointed to either Unity or Godot.

So I downloaded and installed Godot 4.2.1 and started to RTFM. The entire history of my Godot experience over the last 30 days is what's in the Getting Started section of the manual. I started at the beginning of the manual and have slowly been working my way forward. With the hope that Godot would at least teach me about Godot. This entire IDE is new and odd looking to my eye. I don't understand what I'm seeing or how to use it. The fact that I got the player scene working was a moment of rejoicing. The enemy scene is the 1st time in this manual that a 2nd scene in the same project is introduced.

In ASP, each page is its own file since it needed its own address to show different visuals. 
In VB4-VB6, the visuals were a separate thing, and all the code was one long list that referenced itself. My DLL was entirely code with no visual interface whatsoever.
No, I've never programmed in C, C++, or C# to any meaningful extent.

If my past experience of volunteer developers is anything to go by, you live and breathe 1s and 0s and code with an obsession that borders on insanity. Like the hacker that destroyed my PHP projects, you started at a very young age, have an intuitive grasp of programs, and having stared at the Godot interface for the last x years, it make intuitive sense, and how could anyone NOT understand it. 

When a new user, me, cries out for help, I'm told to "respect" you. The history of computers from my original DOS prompt to my Android phone has been a history of "respect" for the customer. As the interface becomes more understood, it becomes more widely used. My FIL is a network administrator and a Linux geek. Linux is stuck in development HELL because they have no respect for the end user. When all I'm trying to do is x, I have to wade through the history of UNIX, the quirky sense of humor of a developer who died in 1969, 4 Lord of the Rings references, and a veiled threat to the IRS because that developer was arrested on tax evasion. All I'm trying to is write my document, see a picture, or play music, not solve the mystery of "Ready Player One".

When I got frustrated w the Getting Started document, it sent me here. It gave no other resources for answers. MOF, a list of resources both within and outside of Godot might be very helpful in the beginning of this doc. 

I'm just trying to understand the interface. The two tabs of "player" and "mob" seem to do nothing. At this point, I have no predictive ability for when I do x, it will do y. Every time I have attempted such a prediction, it's not true. Thus, I'm lost.

I'm not even getting into the nitty-gritty of the project I originally came here to start. This is YOUR project that I'm trying to duplicate. If the musician who wrote the music can't teach you how to play the music that he wrote, why would anyone else know how to play it?

**Comment by BastiaanOlij** [comment]
@TesFalcon, thanks for your life story. I am sorry that you've had such bad experiences in the past. I can assure you, we are nothing like that. I for one have never hacked a computer in my life and while I have respect for people with such skills, I have little respect for kids who destroy someone's efforts just because they can.

For me personally, yes I started programming at a young age, I have a picture of a 6 year old me sitting behind my fathers TRS-80 (might give you a clue of my age), though I didn't really start doing anything serious until my young teens. That might be all that I have in common with your hacker friend. In my teens I was part of a movement of people creating art with computers where sharing your knowledge was paramount. Training others, sharing your knowledge freely, is what I grew up with. I've since worked for decades in enterprise business, not just programming but training both users and new developers, being invited to speak at conferences, etc. Now I work on Godot with a very similar attitude, to spread my knowledge freely, and to help people along.
And that's just my story, amongst the people who work on the documentation you will find teachers on all levels (from teaching primary-schoolers to university students), people who run popular Youtube channels with further training content, alongside many both experienced and new developers from all walks of life.

To paint us all with one brush and call us all hackers that do not care about users, I'm sorry sir, but you are way out of line.

I will not claim that our documentation is perfect, but it has helped thousands of people onboard using Godot. It is the product of years of writing and processing feedback. We welcome further constructive feedback and do continuously work on improving the documentation.
However, there is no such thing as one size fits all documentation, people learn in different ways, and slandering it and the people who wrote it because it doesn't fit you, is not the way to get help nor to suggest improvements.

This is why we're glad that there are other resources available, as shown in the link I already provided (which I admit could be more prominently presented in the manual), and through Youtube channels like the one you already found.
Godot also has a [vibrant user community](https://godotengine.org/community/) on Discord, Reddit, Facebook and other outlets where you will find many people willing to help someone new to Godot jump over any hurdles they encounter.

But if you chose to partake in any of those, I do hope you will enter with a more respectful, less judgemental, attitude than you have shown here so far. Again, I fully understand that we're not at our best when we encounter a frustrating situation, but do try and understand that people here do really want to help.

**Comment by TesFalcon** [comment]
There are 3 edges of technology: 
1) Bleeding edge with the latest and greatest
2) Current where the majority of people are at
3) Trailing edge which is the cheap, used equipment sold by those on the bleeding edge. 

My family has never had been rich. Our 8088 was a hand-me-down PC when a friend of Dad's at work upgraded to the 286. When I arrived at my 1st AFB, they were running 486s while Dad still used that 8088. After his friend at work upgraded to a 486, Dad got his old 286. My 2 sets of golf clubs are also hand-me-downs bought from that same friend whenever he upgraded to the latest and greatest. All my dad knew how to use on either PC we had was Lotus 123. (Talk about a great interface. To bring up the drop-down menu you pressed the slash "\/".) When I introduced him to MS-Office, he rejected it, at first. As he got used to Word, he also started porting all his Lotus spreadsheets to Excel. He used it, but he didn't like it. 

I heard some great things about the TRS-80 from the old developers in the office back in the day, but which edge did you live on: bleeding, current or trailing?

---

It's hard to suggest improvements when you're falling down a well with not even a branch to smack into on the way down. Now that I sit at the bottom of my well of ignorance, I can both say "That 1st step was a doozy" as well as see a place to put a staircase. 

My first assignment when I got to work in the USAF wasn't to actually start programming. I was first assigned to write user documentation for various legacy software that had none. I did that for ~1 yr before the top brass said that if I (and others) who were programmers weren't doing programming work we'd get our job codes changed to match our jobs (operators) rather than stay as programmers. 

In every user documentation that I started or saw written, the 1st step was to describe, in detail, the UI before getting into features or operations. I conducted many interviews with folks who used the interface daily, and they would fly around the various buttons and menu items or use keyboard shortcuts like it was nothing. For a new user, each interface is daunting. MOF, most of our projects were begun because new users were lost, but the folks who knew the system and were supposed to be training them weren't explaining it.

(As I look back over my life, maybe that was a change I shouldn't have made. I enjoyed technical writing. I was doing well there. I never did fit well as a programmer.)

Instead of suggesting a change to the "Creating Enemy" page (which I still don't know how to correct), my 1st suggestion would be a rewrite of the "First Look at Godot's Editor" page. As I'm re-reading that page, it has very little descriptive commentary on each section. I think this is where I first got lost. This section was supposed to be my map to the visual world of Godot, but it's not.

Starting with the Project Manager makes sense since that's the first screen you see when you launch Godot when new.
The !Note block keeps this page short while giving more details for those who need to dig deeper.

The whole menu bar is described in one sentence and a picture then moved on. 
"By default, it features menus, main screens, and playtest buttons along the window's top edge."
Not even a descriptive of each menu item or "main screen" or the types of stuff that I'd find buried in each menu or what the "main screens" are or what playtest buttons do. What are you calling the "main screens"? What does each "main screen" do? If there are multiple "main screens", wouldn't that be a bit of an important feature to highlight?

Where's the !Note block for more info on this?

Next is a reference to a Viewport and Toolbar along with a picture of a 3D space. Is the Viewport one of the "main screens" or is it something different? Why is there no reference to the [:Perspective button] on the left-hand side? 
The references to Toolbars and Docks are out of order. The toolbars are references above the big image, but they aren't discussed or imaged until after the 3D image and the reference to docks.
There's a textual reference to the "docks" and a "bottom panel", but there're no images showing them. ... Correction, the textual reference to the docks is interrupted by images and discussion of the "toolbars" then a return to "docks". SMH.

Again, the docks are imaged, described in one sentence, and moved on. I can understand NOT going into great detail in the beginning of the document with all the features, bells, and whistles of each visual item, but shouldn't there, at least, be a link to a reference with all the nitty-gritty details?

Where's the !Note block for more info on this?

Since the Inspector is such a prominent item that has so much buried in it and is referenced a lot in upcoming pages, more attention should given to the description of the Inspector itself and how it's used. Such as a layout of stuff you're calling "sections" or whatever.

Where's the !Note block for more info on this?

Seriously?! It's all out of order. No wonder I was lost. 

The main screens are down here? I never connected the main screens described down here with the buttons at the top. But, again, these screens are given a big picture, a scant description, and moved on. Not even a link for more info.
 
The Script Screen is another place where predictive functionality is important. The example image only shows one tab. When I click a 2nd tab, nothing seems to change on the script screen, even though that's the tab that I clicked. What is going on?

There's a "!Note" block for more info on the 3D screen, but why not the other screens? There should be a link to more info after every one of these GUI items.
BTW, there's still no reference to Viewport here. 
What is a Viewport and how is that different from a Main Screen?

Asset Lib also has a "more info" block, but why is it called "!See Also" instead of "!Note"? Consistency!

And then this page goes into "Integrated Class References" and with a detailed how-to on finding them. I don't know what these are or what that has to do with what I'm seeing in the GUI.

**Comment by BastiaanOlij** [comment]
Hi @TesFalcon,

Again I have to underline, you are making very base assumptions here. As I mentioned before, Godot has contributors from all walks of life, from countries all over the world. To stay with your example, from all 3 edges of technology. Some who have had more luck than you, some who have had to do with far less. This is one of the things I find beautiful about Godot and its community, its an incredibly empowering tool providing a career in game development to many who otherwise would not have access to such an option.

I understand where you are coming from but I hope you see that Godot is a very different class of software than say late 80ies, early 90ies Lotus 1-2-3 (I too sometimes long back to the simplicity of those days). Many of these applications had a singular function, or had functions divided into singular screens, so explaining the UI elements first works as this is reasonably brief. I have written documentation for plenty of applications such as that myself, I have build many such applications in the past.

But if as you suggest we were to first explain every option you see before you in the Godot IDE, which encompasses all functions of a complete game engine, a new user would have to read through pages and pages and pages of information that is completely meaningless to that user because they are new and have no frame of reference to make sense of the information provided. They would give up on Godot long before they create their first game.

Instead we have chosen for a high level overview so the user has an idea where to look, and then get to building a simple game first. As they get more skilled with the engine, more and more relevant things are explained. When they get stuck, there is a vibrant community of users willing to help out and get them over hurdles.

Godot simply put has a very different class of interface and you will find that other software that shares that trait, will have a similar approach in their documentation and in supporting their user base.

Again, I highly suggest to you to have a look at the links I have already provided and to make use of the community outlets I mentioned before.

The issue list here on GitHub is not the right place to find user support. If we have given that impression than that is valuable feedback that we will take action upon and are already discussing internally.

As I am again circling back to the same advise I have provided in 3 reactions now, I will leave it there.

**Comment by TesFalcon** [comment]
You said, "I have a picture of a 6 year old me sitting behind my fathers TRS-80 (might give you a clue of my age)"

The TRS-80 was released in 1972. At the oldest, you were born in the late 60s if your father was on the bleeding edge of technology. If he was not or if the TRS-80 was pulled out of storage like the 286 my FIL pulled out w a Linux distro for my kids in 2008, then you could be nearly any age younger than that. Our family's 8088 was acquired in 1990; that's a LONG time after it was released in 1977. As my personal stories continued, I gave example after example of Dad being a laggard in tech, not on the bleeding edge. He got a 286 in 1995. He preferred Lotus 123 even when MS-Excel was dominating the spreadsheet space in offices worldwide. He started learning about Windows 3.1 around the time that Win95 was released. 

How you managed to insinuate a connection between Lotus123 and Godot, I have NO idea. 

I used the bars to designate the change between topics.

Idea #1
---
Idea #2
 
Apparently you didn't actually read what I wrote because I did talk about the need to balance brevity with depth. 4x I asked "Where's the !Note block for this?" IOW, if you want to stay as brief as you are here, there should be more available for those who want to dig deeper. Back when software came in boxes and the user's manual was a full pound of paper or more, I often found this sort of in-depth scrolls through the menu tree along with images and description in the appendix of the book. At the front, there'd be a short-hand reference, but then there's the appendix reference. 

("You can change settings in the Settings Menu. For more details on the Settings Menu, see Appendix 6.4.")

Between this poorly written "Getting Started" section and this conversation, I am fully ready to "give up on Godot long before they create their first game." If you have forgotten, this entire conversation is my attempt to follow Godot's own recipe for "Your 1st 2D Game". It's not even a game of my design or interest. If it not for GDQuest, I still might. We'll see how that goes. 

---
Idea #3

"the user has an idea where to look": The manual pointed me HERE. I didn't find this chat by Googling something. The manual pointed me here to discuss issues with the manual. It's Godot's own cookbook recipe that I'm complaining about. If Betty Crocker's recipe is flawed, you discuss that w Betty Crocker, not Julia Child. Cooking is a nearly perfect metaphor for this since all cookbooks make certain assumptions about your level of skill. A thick cookbook (300+ pages) has a lot of reference pages in the front to explain the measurements, French terms, etc. They'll even have an example recipe with a description of how each recipe is laid out and where to find more info (in that cookbook) on things you don't know. They assume that you don't know how to cook. A small cookbook pamphlet that you might find at the grocery store ignores all of that because they assume that you know the difference between dicing vs mincing, sauteeing vs brazing, etc. and just get to the 50 Irish Stew recipes that you came for with a total page count of 52.

---
Idea #4

"Godot simply put has a very different class of interface" 

What "class of interface" is it? 

"you will find that other software that shares that trait, will have a similar approach in their documentation and in supporting their user base."

Are you assuming I am familiar with other software in this class? 
Do the other software in this class similarly ignore, overlook, and disdain their new users, expecting someone else to help them since they did such a poor job of explaining their own interface?

--- 
Idea #5

"The issue list here on GitHub is not the right place to find user support. If we have given that impression than that is valuable feedback that we will take action upon and are already discussing internally."

And not a single mention is made of the numerous items that I mentioned of the page being poorly ORDERED. To phrase it in a more programmatic style, "First look at Godot's editor" is spaghetti code.

**Comment by BastiaanOlij** [comment]
Hi @tesfalcon,

I did actually read through all your posts fully but I hope you understand that due to their length and long tangents about your personal life, most people will give up.

I will offer up these final thoughts.

First, I took some time yesterday to revisit your earlier post and my remark that we may have missed informing users clearly where to find help, turns out to be wrong.
The very first page of the manual starts with a nice welcome message immediately followed by telling you where to find tutorials and where to ask for help, and I quote:

> In case you have trouble with one of the tutorials or your project, you can find help on the various [Community channels](https://docs.godotengine.org/en/stable/community/channels.html#doc-community-channels), especially the Godot [Discord](https://discord.gg/4JBkykG) community and [Forum](https://forum.godotengine.org/).

The last paragraph on this very first page provides further information about the maintenance of this documentation. I did not find reference to your opinion that we sent you here to find help.

Second, going through your suggestions in your early post side by side, the additions you suggest would make the documentation needlessly verbose, it distracts the user from what is important at this early stage and would needlessly confuse them.

This drip feeding of information as you get more familiar with an application, especially one as many faceted as Godot, is a tried and tested staple in the industry.

Have we got it perfected? Certaintly not, but your current approach is not conductive to improving this and again, slandering the work of countless people, and then insulting them by calling them hackers or insinuating we don't understand our users because you think we have a more privileged life than you is inexcusable. This is not how you solicit help.

I once again implore you to use the community outlets available to you to find help so you can complete the tutorial and get some miles under your belt. You may find that the ingredient you missed is small and requires a short, to the point, change to the documentation to prevent others from stumbling on the same problem and we will welcome that suggestion. One you may very well have already suggested but which is currently lost among the pages and pages of frustration.

As we are now fully going around in circles, we are closing this discussion.

I wish you well and I truly hope you will follow our suggestions and find the answer you are looking for.
